src/core/src/node.cpp

```diff
@@ -272,9 +272,13 @@ void ov::Node::set_input_is_relevant_to_value(size_t i, bool relevant) {
 }
 
 void ov::Node::set_output_type(size_t i, const element::Type& element_type, const PartialShape& pshape) {
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    get_output_descriptor(i).get_tensor_ptr()->set_tensor_type(element_type, pshape);
-    OPENVINO_SUPPRESS_DEPRECATED_END
+    auto tmp_tensor = std::make_shared<ov::descriptor::Tensor>(element_type,
+                                                               pshape,
+                                                               get_output_descriptor(i).get_tensor_ptr()->get_names());
+    for (auto& item : get_output_descriptor(i).get_tensor_ptr()->get_rt_info()) {
+        tmp_tensor->get_rt_info()[item.first] = item.second;
+    }
+    get_output_descriptor(i).set_tensor_ptr(tmp_tensor);
```

Comment:
@ilya-lavrenov  @vurusovs  Why we need to deprecated this API? Seems like this change will take more time & more complex.